### ISSUE TITLE
5.x not removing fields via grok mutate remove_fields statement

### ISSUE THREAD

**Issue by xerxes9000sx** [elastic/logstash#6384]
I remove tags field via mutate remove_fields statement. 
Example logstash config:
```
input { stdin { tags => ["test"] } }
filter { if "test" in [tags] { mutate { remove_field => [ "tags" ] } } }
output { stdout { codec => rubydebug } }
```

2.4.1 & 5.0.0 removing filed tags, later versions leave filed empty.
If I add --debug switch I can see any version try to remove tags:
```
[2016-12-09T14:42:37,895][DEBUG][logstash.util.decorators ] inputs/LogStash::Inputs::Stdin: adding tag {"tag"=>"test"}
[2016-12-09T14:42:37,906][DEBUG][logstash.pipeline        ] filter received {"event"=>{"@timestamp"=>2016-12-09T11:42:37.892Z, "@version"=>"1", "host"=>"vanilin", "message"=>"testing", "tags"=>["test"]}}
[2016-12-09T14:42:37,908][DEBUG][logstash.filters.mutate  ] filters/LogStash::Filters::Mutate: removing field {:field=>"tags"}
[2016-12-09T14:42:37,910][DEBUG][logstash.pipeline        ] output received {"event"=>{"@timestamp"=>2016-12-09T11:42:37.892Z, "@version"=>"1", "host"=>"vanilin", "message"=>"testing", "tags"=>[]}}
{
    "@timestamp" => 2016-12-09T11:42:37.892Z,
      "@version" => "1",
          "host" => "vanilin",
       "message" => "testing",
          "tags" => []
}
```

**Comment by colinsurprenant** [comment]
confirmed and duplicate of #6429 
pushed fix #6477

**Comment by colinsurprenant** [comment]
fixed by #6477, will be included in 5.1.2

**Comment by colinsurprenant** [comment]
thanks @xerxes9000sx for reporting!
